File: frontend/src/types/investigation.ts
export type RiskLevel = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';
export type EntityType = 'SOCIAL_PROFILE' | 'WEBSITE';
export type SignalAvailability = 'AVAILABLE' | 'UNAVAILABLE';

export interface AnalysisSignal {
  id: string;
  signal_name: string;
  signal_category: string;
  detected: boolean;
  weight: number;
  value?: string | null;
  explanation: string;
  availability: SignalAvailability;
  created_at: string;
}

export interface EntityConnection {
  id: string;
  source_entity_id: string;
  target_entity_id: string;
  connection_type: string;
  connection_reason: string;
  similarity_score: number;
  target_domain?: string | null;
  target_platform?: string | null;
  created_at: string;
}

export interface InvestigationListItem {
  id: string;
  original_url: string;
  normalized_url: string;
  entity_type: EntityType;
  domain: string;
  platform: string;
  status: string;
  risk_score: number;
  risk_level: RiskLevel;
  summary?: string | null;
  created_at: string;
  completed_at?: string | null;
}

export interface SocialProfileData {
  id?: number;
  username: string;
  display_name?: string | null;
  platform: string;
  profile_url?: string | null;
  profile_image_url?: string | null;
  bio?: string | null;
  followers_count?: number | null;
  following_count?: number | null;
  posts_count?: number | null;
  availability?: Record<string, string>;
  status_message?: string | null;
}

export interface InvestigationDetail extends InvestigationListItem {
  signals: AnalysisSignal[];
  connections: EntityConnection[];
  profile?: SocialProfileData | null;
}


export interface PaginatedInvestigations {
  total: number;
  page: number;
  limit: number;
  pages: number;
  items: InvestigationListItem[];
}

File: backend/app/schemas/investigation.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, Field

class InvestigationCreate(BaseModel):
    url: str = Field(..., description="The URL to submit for analysis")

class SignalResponse(BaseModel):
    id: str
    signal_name: str
    signal_category: str
    detected: bool
    weight: int
    value: Optional[str] = None
    explanation: str
    availability: str
    created_at: datetime

    class Config:
        from_attributes = True

class ConnectionResponse(BaseModel):
    id: str
    source_entity_id: str
    target_entity_id: str
    connection_type: str
    connection_reason: str
    similarity_score: float
    target_domain: Optional[str] = None
    target_platform: Optional[str] = None
    created_at: datetime

    class Config:
        from_attributes = True

class InvestigationListItem(BaseModel):
    id: str
    original_url: str
    normalized_url: str
    entity_type: str
    domain: str
    platform: str
    status: str
    risk_score: int
    risk_level: str
    summary: Optional[str] = None
    created_at: datetime
    completed_at: Optional[datetime] = None

    class Config:
        from_attributes = True

class InvestigationDetail(InvestigationListItem):
    signals: List[SignalResponse] = []
    connections: List[ConnectionResponse] = []
    profile: Optional[dict] = None

InvestigationDetailResponse = InvestigationDetail


class PaginatedInvestigations(BaseModel):
    total: int
    page: int
    limit: int
    pages: int
    items: List[InvestigationListItem]
